controllers/application.go

```diff
@@ -110,6 +110,14 @@ func (c *ApiController) GetApplication() {
 		}
 	}
 
+	if application != nil {
```

Comment:
we should just check api that used to login / signup, now, we should check ip in `Signup() and Login()`

Reply:
This is to check whether the user's IP meets the rules directly through the application and organization's IP whitelist when the user visits the entry page.
It is equivalent to a pre-check. Compared with this check, the check in login and signup has an additional user's IP whitelist.

Reply:
but when we use GetApplication as a restful api, maybe we don't need whitelist function, we just want to check ip in entryPage

Reply:
Could you please explain the restful api in detail?
My current logic is: if the user's IP is not in the whitelist, when it accesses the entryPage to log in, it will not be able to continue to access other pages because of the IP forbidden error returned by the get-application request. In other words, when the user's IP is not in the whitelist, this API will only be used by the entryPage.
So I think we don't need to consider that other scenarios that will use this API will be affected by the IP forbidden error, because when your IP is banned and cannot access the entryPage, you should not be able to continue to access our services.

Reply:
In my first implementation commit:05c098ff1b690a23e6e131f2450430ab5391f362, I added a check-entry-ip API to check whether the IP meets the whitelist of application and organization.
The check-entry-ip API is only used in LoginPage, SignupPage and ForgetPage, and will be called after the get-application API succeeds. When check-entry-ip returns an error message, the page will jump to the error page.
I think this implementation method may meet your needs, but on the one hand, it makes too many changes to the front end (compared to the current modification of the display method of get-application errors); on the other hand, it brings additional request APIs.

The practice of adding entryIp check to get-application can realize IP check for application and organization whitelists on the one hand, and on the other hand, although this will affect casdoor-sdk's direct call to get-application, I think this impact is tolerable when users cannot log in.

Because if the user can't even log in, there's no point in him calling get-application. And if he wants to check why he can't log in, our error message also gives the specific level (organization/application) and specific name of the whitelist.

Reply:
check ip in CRUD is useless and may cause some issue, you can use a external parameter to decide whether check IP in `GetApplication` interface.

Reply:
@ZhaoYP-2001 you should add `bool IsRestricted` field to Application and Organization. `IsRestricted` will be assigned in the Go's REST API handler. Frontend will respect this field and show error if needed